### PR TITLE
Roy kanban

### PR DESCRIPTION
**Pull request by RoyShemesh**
This is my pre-course final project!

### CODE REVIEW

File: solution/index.html [hunk 1]
```diff
@@ -5,9 +5,73 @@
...
+              <button onclick="nextBg()" style="width:5%"class="carousel-control-next" type="button" data-bs-target="#carouselExampleCaptions" data-bs-slide="next">
+                <span class="carousel-control-next-icon" aria-hidden="true"></span>
+                <span class="visually-hidden">Next</span>
+              </button>
+            </div>
+    </header>
+    <main id="main"style="display: flex;">
+        <section class="box" id="1" >
+            <h4 class="display-4">To do</h4>
+            <ul class="to-do-tasks" id="todo" ondragover="allowDrop(event)" ondrop="drop(event)">
```
Comment (ofer3k) [inline solution/index.html:39]: Uses event listener functions in a way that is explained as not optimal

File: solution/index.js [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+async function getApi(){
+    showSpinner();
+    let alltask=document.getElementsByTagName("li");
+    for (let i = alltask.length-1; i >-1; i--) {
+        alltask[i].parentElement.removeChild(alltask[i]);        
+    }
+    try
+    {
+        const respone= await fetch("https://json-bins.herokuapp.com/bin/614b080c4021ac0e6c080cd2",{
```
Comment (langerapp) [inline solution/index.js:47]: wrong spacing coding style

File: solution/index.js [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+                task.addEventListener('mouseout', moveTaskEnd);
+            });
+        }
+    }
+    finally{
+        hideSpinner();
+    }   
+}
+async function saveApi()
+{
```
Comment (langerapp) [inline solution/index.js:93]: separate between network logic & ui logic

File: solution/index.js [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+            check=false;
+        }
+    }
+    localStorage.setItem("tasks",JSON.stringify(tasksJson));
+}
+
+
+
+//add and remove key press event
+let moveTaskLi;
```
Comment (TswusayonaTsamsiyu) [inline solution/index.js:277]: You don't need a global variable to know which task element is hovered. Find another way

File: solution/index.js [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+        list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+        list["in-progress"].unshift(moveTaskLi.innerText);
+        localStorage.setItem("tasks",JSON.stringify(list));
+    }
+    if(event.keyCode===51&&event.altKey)
+    {        
+        document.getElementById("done").prepend(moveTaskLi)
+        list=JSON.parse(localStorage.getItem("tasks"));//push the task to local storage
+        list["done"].unshift(moveTaskLi.innerText);
+        localStorage.setItem("tasks",JSON.stringify(list));
```
Comment (TswusayonaTsamsiyu) [inline solution/index.js:314]: There's a lot of copy-paste here. Create more helper functions

**Pull request merged**